wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -811,9 +812,24 @@ public final class SessionManager: NSObject, SessionManagerType {
                     if error != nil {
                         self.delegate?.sessionManagerDidFailToLoadDatabase()
                     } else {
-                        let userSession = self.startBackgroundSession(for: account,
-                                                                         with: coreDataStack)
-                        completion(userSession)
+                        let userSession = self.startBackgroundSession(for: account, with: coreDataStack)
+
+                        if self.cryptoboxMigrationManager.isNeeded(in: coreDataStack.accountContainer) {
```

Comment:
I just have a concern related to where we check and start the migration. There's an `if else` statement starting at line 797 which looks like this

```
if let session = self.backgroundUserSessions[account.userIdentifier] {
    // session already loaded. call completion handler with session
} else {
    // create session
}
```

The migration happens in the `else` block. So I wonder what happens when there are sessions already loaded. Do we not migrate them ?

Reply:
As discussed offline, we only need to migrate once when the session is first loaded. Since sessions are kept in memory, each session must be loaded at least once during the lifetime of the app. Since the app will upgrade, it will need to be launched.